packages/overlay/src/ActiveOverlay.ts

```diff
@@ -392,37 +372,15 @@ export class ActiveOverlay extends SpectrumElement {
 
     private returnOverlayContent(): void {
         /* c8 ignore next */
-        if (!this.overlayContent) return;
+        if (!this.restoreContent) return;
 
-        if (this.originalSlot) {
-            this.overlayContent.setAttribute('slot', this.originalSlot);
-            delete this.originalSlot;
-        } else {
-            this.overlayContent.removeAttribute('slot');
-        }
-
-        if (this.placeholder) {
-            const parentElement =
-                this.placeholder.parentElement ||
-                this.placeholder.getRootNode();
-            if (parentElement) {
-                parentElement.replaceChild(
-                    this.overlayContent,
-                    this.placeholder
-                );
-                this.willNotifyClosed = true;
-            }
-        }
+        const [element] = this.restoreContent();
+        this.restoreContent = undefined;
 
         if (this.originalPlacement) {
-            this.overlayContent.setAttribute(
-                'placement',
-                this.originalPlacement
-            );
+            element.setAttribute('placement', this.originalPlacement);
```

Comment:
This is managed here, and `slot` is managed there... it begs a question should the API be:
```
reparentChildren(Element[], Element, () => () => void);
```
Where the double callback at the end amount to `prepare` and `cleanup` functionality?

In this way, `slot` wouldn't have to be managed on _everything_, and `placement` or whatever else came up over time, wouldn't have to be cached in the originating code location. 

Everything could just be managed/cached at the originating location, just the same, but want to make the workflow clear as one of the follow up tasks to the overall refactor here will be to add this method to: https://opensource.adobe.com/spectrum-web-components/components/shared

Reply:
I'm assuming the callbacks would be optional -- definitely makes sense to add more control to the API.

Reply:
Check it out in #1310